File: algorithms/FooBar/FooBar.kt
fun main()
{
    var start: Int;
    val foo: Int = 5;
    val bar: Int = 7;
    var stop: Int;
    val fooString: String = "Foo";
    val barString: String = "Bar";
    start = 1;
    stop = 100;
    
    for (num in start..stop)
    {
        if (num % foo == 0 && num % bar == 0)
        {
            println(fooString + barString);
        }
        else if (num % foo == 0)
        {
            println(fooString)
        }
        else if (num % bar == 0)
        {
            println(barString)
        }
        else
        {
            println(num);
        }
    }
}

File: algorithms/FooBar/fooBar.go
package main

import("fmt")

func main() {
	start, foo, bar, stop, num := 1, 5, 7, 100, 0;
	fooString, barString := "Foo", "Bar";
	for num = start; num <= stop; num++ {
		if num % foo == 0 && num % bar == 0 {
			fmt.Printf(fooString + barString + "\n");
		} else if num % foo == 0 {
			fmt.Printf(fooString + "\n");
		} else if num % bar == 0 {
			fmt.Printf(barString + "\n");
		} else {
			fmt.Printf("%d\n", num);
		}
	}
}
